tools/plugins/com.liferay.ide.project.ui/src/com/liferay/ide/project/ui/languageserver/LanguageServerConnectionProvider.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.ide.project.ui.languageserver;
+
+import com.liferay.ide.core.util.CoreUtil;
+import com.liferay.ide.project.core.modules.BladeCLIException;
+
+import java.io.File;
+
+import java.util.Arrays;
+import java.util.Properties;
+
+import org.eclipse.lsp4e.server.ProcessOverSocketStreamConnectionProvider;
+import org.eclipse.wst.server.core.util.SocketUtil;
+
+/**
+ * @author Terry Jia
+ */
+public class LanguageServerConnectionProvider extends ProcessOverSocketStreamConnectionProvider {
+
+	public LanguageServerConnectionProvider() throws BladeCLIException {
+		super(
+			Arrays.asList("java", "-DliferayLanguageServerPort=" + _port, "-jar", _getLiferayPropertiesServerPath()),
```

Comment:
@jtydhr88  @Seiphon   HOw does LSP4XML and other projects in eclipse handle the launching of the LSP process?  DO they also use the `ProcessOverSocketStreamConnectionProvider` class? Do they not use the JRE launching facilities in Eclipse (to launch a long running JRE server process? )

Reply:
LSP4XML is using the same way to launch LSP process, they use "ProcessStreamConnectionProvider" class, and we will it replace "ProcessOverSocketStreamConnectionProvider" later, because we will use standard io replace socket to communicate.

And LSP is not a long running JRE server process, its lifecycle is from openning a properties file to close this file.